spec/controllers/idv/confirmations_controller_spec.rb

```diff
@@ -25,34 +25,34 @@
     end
 
     describe 'all questions answered correctly' do
-      it 'shows success' do
+      it 'updates profile and redirects to SP' do
         complete_idv_session(true)
 
         get :index
 
-        expect(response.status).to eq 200
-        expect(response.body).to include(t('idv.titles.complete'))
+        expect(response).to redirect_to('http://the-original-sp.example.com')
+        expect(flash[:success]).to eq(t('idv.titles.complete'))
 
         profile.reload
 
         expect(profile).to be_active
         expect(profile).to be_verified
+
+        # PII removed from session
+        expect(subject.user_session[:idv][:applicant]).to eq nil
```

Comment:
Yes, I see what is being tested, and it is valid except for the description and URL. The user is not redirected to the SP at that point:
`subject.session[:saml_request_url] = 'http://the-original-sp.example.com'`
is overriding the value for the [originating SAML request](https://github.com/18F/identity-idp/blob/master/app/controllers/concerns/saml_idp_auth_concern.rb#L21) and not a destination at the SP.

The spec suggests that the user is redirected to the SP upon completion of Identity Verification, which is true-ish. The flow is:
1. Authn request to /api/saml/auth 
2. User redirected to sign in and 2FA
3. User redirected to Identity Verification
4. Now that the user has achieved the requested LOA, the AuthnRequest can now be handled; we need to test for a valid Assertion, or, if the SAMLRequest has expired in the meantime, we need to forward back to SP to generate a fresh AuthnRequest.